irc.go

```diff
@@ -125,9 +125,13 @@ func parseIRCMessageSource(rawSource string) *ircMessageSource {
 	regex := regexp.MustCompile(`!|@`)
 	split := regex.Split(rawSource, -1)
 
-	if len(split) == 1 {
-		source.Host = split[0]
-	} else {
+	if len(split) == 1 || len(split) == 2 {
```

Comment:
This should be split up into a "if len == 1, else if len == 2, else if len >= 3" case, and in the len == 2 case, we should populate Nickname with split[0] and Host with split[1]